lib/src/arc_text_painter.dart

```diff
@@ -13,40 +13,52 @@ class ArcTextPainter {
     double initialAngle = 0,
     Direction direction = Direction.clockwise,
     Placement placement = Placement.outside,
+    double stretchAngle,
+    double interLetterAngle,
   })  : assert(radius != null, 'radius should not be null'),
         assert(text != null, 'text should not be null'),
         assert(textStyle != null, 'textStyle should not be null'),
         assert(alignment != null, 'alignment should not be null'),
         assert(initialAngle != null, 'initialAngle should not be null'),
         assert(direction != null, 'direction should not be null'),
         assert(placement != null, 'placement should not be null'),
+        assert(!(stretchAngle != null && interLetterAngle != null),
+            'stretchAngle and interLetterAngle should not be both not null'),
         this._text = text,
         this._textStyle = textStyle {
     _textPainter.text = TextSpan(text: _text, style: textStyle);
     _textPainter.layout(minWidth: 0, maxWidth: double.maxFinite);
 
     switch (placement) {
       case Placement.inside:
-        this._radius = radius;
+        this._radius = radius - _textPainter.height;
         break;
       case Placement.outside:
-        this._radius = radius + _textPainter.height;
+        this._radius = radius;
         break;
     }
 
+    _interLetterAngle = interLetterAngle ?? 0;
+    final double finalAngle = getFinalAngle();
+    final double alignmentOffset = _getAlignmentOffset(
+        alignment, stretchAngle != null ? stretchAngle : finalAngle);
     switch (direction) {
       case Direction.clockwise:
-        _angleWithAlignment = initialAngle + _getAlignmentOffset(alignment);
+        _angleWithAlignment = initialAngle + alignmentOffset;
         _angleMultiplier = 1;
-        _heightOffset = -this._radius;
+        _heightOffset = -this._radius - _textPainter.height;
         break;
       case Direction.counterClockwise:
-        _angleWithAlignment =
-            initialAngle - _getAlignmentOffset(alignment) + math.pi;
+        _angleWithAlignment = initialAngle - alignmentOffset + math.pi;
         _angleMultiplier = -1;
-        _heightOffset = this._radius - _textPainter.height;
+        _heightOffset = this._radius;
         break;
     }
+
+    if (stretchAngle != null && _text.runes.length > 1) {
+      _interLetterAngle =
+          (stretchAngle - finalAngle) / (_text.runes.length - 1);
```

Comment:
```suggestion
      _interLetterAngle = (stretchAngle - finalAngle) / _text.runes.length;
```

I think without `-1` it looks better, at least with angle `360` it seems more natural:

![image](https://user-images.githubusercontent.com/853356/83360709-94294f80-a383-11ea-916f-55b7a12f2f09.png)